src/main/java/com/school/project/service/impl/UserAccountServiceImpl.java

```diff
@@ -23,4 +26,20 @@ public UserAccount createUserAccount(User user, UserAccountType userAccountType)
         userAccountRepository.save(userAccount);
         return userAccount;
     }
+
+    @Override
+    public UserAccount getUserAccountByUserId(Long id) {
+        Optional<UserAccount> account = userAccountRepository.getUserAccountByUserId(id);
+        if (!account.isPresent()) throw new UserAccountNotFoundException("User account with userId : " + id + " is not found");
+        return account.get();
+    }
+
+    @Override
+    public void updateAccountRole(Long userId, Integer statusId){
+        UserAccount account = getUserAccountByUserId(userId);
+        account.setAccountRole(UserAccountType.getTypeById(statusId));
```

Comment:
I think you have kinda logic mistake here. The default enum converter takes all the enums of one class in sort of a list with ids, where id 0 will return first enum. So if you will send status code 1, it will be second account type enum and etc. your function UserAccountType.getTypeById() should reflect this logic, otherwise you will assign wrong account types.